### PR TITLE
API endpoints for reset password flow

### PR DESCRIPTION
**Pull request by baylee**
@rmutter take a look. Moves password reset stuff into API endpoints instead of the standard Django flow.

### CODE REVIEW

File: test_project/api-schema-1.0.json [hunk 1]
```diff
@@ -256,7 +256,7 @@
             }
         },
         {
-            "url": "users/:id/set_password",
+            "url": "password/change/",
```
Comment (baylee) [inline test_project/api-schema-1.0.json]: @rmutter breaking change

File: test_project/api-schema-1.0.json [hunk 2]
```diff
@@ -504,10 +526,26 @@
         }},
         {"$changePasswordRequest": {
             "old_password": "string",
-            "password": "string"
+            "password": "string",
+            "confirm_password": "string"
```
Comment (baylee) [inline test_project/api-schema-1.0.json]: @rmutter breaking change

**Comment by rmutter** [comment]
@baylee should the server worry about having 2 passwords to confirm or should that just be a front end validation?

**Comment by baylee** [comment]
@rmutter just figured it was safer to check on the front and the back, in case we ever forget to implement on the front

**Comment by rmutter** [comment]
@baylee okay, looks like either we could just remove it, since it seems a bit unnecessary and breaks existing functionality or we can keep it, but looks like it needs a test.

**Pull request merged**